apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/TimerInterceptor.java

```diff
@@ -0,0 +1,37 @@
+package gov.cms.bfd.server.war;
+
+import ca.uhn.fhir.interceptor.api.Hook;
+import ca.uhn.fhir.interceptor.api.Interceptor;
+import ca.uhn.fhir.interceptor.api.Pointcut;
+import ca.uhn.fhir.rest.server.servlet.ServletRequestDetails;
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+import java.sql.Timestamp;
+import org.slf4j.MDC;
+
+@Interceptor
+public class TimerInterceptor {
+
+  @Hook(Pointcut.SERVER_INCOMING_REQUEST_PRE_HANDLED)
+  public void requestPreHandled(ServletRequestDetails theRequestDetails) {
+    // Record the time before the BFD Handler is called.
+    Long requestStartMilliseconds =
+        (Long) theRequestDetails.getServletRequest().getAttribute(BfdMDC.REQUEST_START_KEY);
+    if (requestStartMilliseconds != null) {
+      // convert requestStartMilliseconds to timestamp
+      Timestamp requestStart = new Timestamp(requestStartMilliseconds);
+      MDC.put(
+          "http_access.request.server_incoming_request_pre_handle_time", requestStart.toString());
```

Comment:
I would like to change the name of this to make it more clear where it came from. proposing 'hapi. server_incoming_request_pre_handle_time' (similar comment on the other one)

Reply:
Sounds good